### ISSUE TITLE
NullPointerException thrown by BeamCommitLogReader.getWatermark

### ISSUE THREAD

**Issue by tisonet** [O2-Czech-Republic/proxima-platform#165]
```
2020-07-27 14:37:18
org.apache.flink.streaming.runtime.tasks.AsynchronousException: Caught exception while processing timer.
	at org.apache.flink.streaming.runtime.tasks.StreamTask$StreamTaskAsyncExceptionHandler.handleAsyncException(StreamTask.java:1073)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.handleAsyncException(StreamTask.java:1041)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invokeProcessingTimeCallback(StreamTask.java:1503)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$null$10(StreamTask.java:1492)
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$SynchronizedStreamTaskActionExecutor.run(StreamTaskActionExecutor.java:87)
	at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:78)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:261)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:186)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:485)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:469)
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:708)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:533)
	at java.lang.Thread.run(Thread.java:748)
Caused by: TimerException{java.lang.NullPointerException}
	... 11 more
Caused by: java.lang.NullPointerException
	at cz.o2.proxima.beam.direct.io.BeamCommitLogReader.getWatermark(BeamCommitLogReader.java:378)
	at cz.o2.proxima.beam.direct.io.BeamCommitLogReader.access$400(BeamCommitLogReader.java:46)
	at cz.o2.proxima.beam.direct.io.BeamCommitLogReader$UnboundedCommitLogReader.getWatermark(BeamCommitLogReader.java:108)
	at org.apache.beam.runners.flink.translation.wrappers.streaming.io.UnboundedSourceWrapper.onProcessingTime(UnboundedSourceWrapper.java:452)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invokeProcessingTimeCallback(StreamTask.java:1501)
	... 10 more
```

**Comment by dmvk** [comment]
Thanks for the report. This is due to race condition in Beam's `UnboundedSourceWrapper`, which under some circumstances (expensive call to Reader#start() and multiple localReaders), can break Reader's contract and call `#getWatermark()` method before `#start()`. This is due to watermark being update asynchronously in processing time timer.

https://github.com/apache/beam/blob/master/runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java#L448

**Comment by dmvk** [comment]
We can hotfix this on our side, by returing min watermark when `observer` is not yet initialized. This should also be fixed in Beam master.

**Comment by je-ik** [comment]
Closed via datadrivencz#412
